config/config.go

```diff
@@ -144,20 +144,35 @@ type Gen2Config struct {
 
 // VPCProviderConfig configures a specific instance of a VPC provider (e.g. GT/GC/Z)
 type VPCProviderConfig struct {
-	Enabled              bool   `toml:"vpc_enabled" envconfig:"VPC_ENABLED"`
+	Enabled bool `toml:"vpc_enabled" envconfig:"VPC_ENABLED"`
+
+	//valid values (gc|g2), if unspecified, GC will take precedence(if both are specified)
+	//during e2e test, user can specify its own preferred type during execution
+	VPCTypeEnabled       string `toml:"vpc_type_enabled" envconfig:"VPC_TYPE_ENABLED"`
 	VPCBlockProviderName string `toml:"vpc_block_provider_name" envconfig:"VPC_BLOCK_PROVIDER_NAME"`
-	EndpointURL          string `toml:"gc_riaas_endpoint_url"`
-	TokenExchangeURL     string `toml:"gc_token_exchange_endpoint_url"`
-	APIKey               string `toml:"gc_api_key" json:"-"`
-	Encryption           bool   `toml:"encryption"`
-	ResourceGroupID      string `toml:"gc_resource_group_id"`
-	VPCTimeout           string `toml:"vpc_api_timeout,omitempty" envconfig:"VPC_API_TIMEOUT"`
-	MaxRetryAttempt      int    `toml:"max_retry_attempt,omitempty" envconfig:"VPC_RETRY_ATTEMPT"`
-	MaxRetryGap          int    `toml:"max_retry_gap,omitempty" envconfig:"VPC_RETRY_INTERVAL"`
-	VPCAPIGeneration     int    `toml:"vpc_api_generation" envconfig:"VPC_API_GENERATION"`
-
-	APIVersion string `toml:"api_version,omitempty" envconfig:"VPC_API_VERSION"`
-	IsIKS      bool   `toml:"is_iks,omitempty"`
+	VPCBlockProviderType string `toml:"provider_type"`
```

Comment:
from where we are reading this and how we are updating ?

Reply:
**This comes part of secret in DEV cluster (may network team is putting that, thats good only)**
$ kubectl get secret storage-secret-store -n kube-system -o yaml | grep slclient.toml: | awk '{print $2}' | base64 --decode

[VPC]
  g2_token_exchange_endpoint_url = "https://iam.bluemix.net"
  g2_riaas_endpoint_url = "https://us-south.iaas.cloud.ibm.com"
  g2_resource_group_id = "51c1427b084f4799abf4f61f531abf76"
  g2_api_key =XXXX"
  encryption = true
  **provider_type = "g2"**